SR2024 - Update Section - Motor board API (#494)
srobo/docs: programming/motors.md

Motor Board API
===============

> The kit can control multiple motors simultaneously

Reviewer: ```suggestion
The kit can control multiple motors simultaneously.
```

Author: a40db43

Each Motor Board can control two motors.
See the [Motor Board](/docs/kit/motor_board) hardware page for more details.

Accessing the Motor Board
-------------------------

If there is exactly one Motor Board connected to your robot, it can be accessed using the `motor_board` property of the `Robot` object.

~~~~~ python
from sr.robot3 import *
robot = Robot()

my_motor_board = robot.motor_board
~~~~~

If you have more than one Motor Board attached, you need to specify which one you want to control.
This is done using the serial number of the board.

The serial number is physically written on each board and the serial of any connected board will be printed to your log when the robot starts.
It will look something like this:

~~~~~ not-code

> sr.robot3.robot - INFO - Found MotorBoard, serial: srABC1

Reviewer 2: Do we want log lines from both the boards below to be consistent?

Author: Good idea - 61ebb2c

~~~~~

You can then access the boards like this:

~~~~~ python
from sr.robot3 import *
robot = Robot()

my_motor_board = robot.motor_boards["srABC1"]
my_other_motor_board = robot.motor_boards["srXYZ1"]
~~~~~

<div class="warning" markdown="1">

>   When you have more than one Motor Board connected to your kit, you must use `robot.motor_boards` and index by serial number.

Reviewer: Nit: Be specific that the issue is not using `robot.motor_board` vs needing to use `robot.motor_boards`.

```suggestion
  When you have more than one Motor Board connected to your kit, you can't use `robot.motor_board`.
```

Author: a40db43

~~~~~ python
# Set one motor to full power in one direction and
# another to full power in the other:
robot.motor_boards["srABC1"].motors[0].power = 1
robot.motor_boards["srABC1"].motors[1].power = -1

# Wait a while for the robot to move
time.sleep(3)

# Stop both motors
robot.motor_boards["srABC1"].motors[0].power = 0
robot.motor_boards["srABC1"].motors[1].power = 0
~~~~~

<div class="info" markdown="1">
  You will need to work out for your robot which values (positive or negative) result in it moving in each direction.

>   This will depend on how you have positioned your motors as well as how they have been wired to the Motor Board.

Reviewer: Suggestion: Perhaps a tip here that it's likely easier to swap the wiring than having motors spin in different directions, and that it's easier for humans to have positive values be forward and negative backwards.

Author: a40db43
